cache/slru_cache.cpp

```diff
@@ -242,66 +171,84 @@ int slru_cache_t::write(const unsigned char *id, dnet_net_state *st, dnet_cmd *c
 	m_cache_stats.size_of_objects -= it->size();
 
 	TIMER_START("write.modify");
-	if (append) {
-		raw.data().insert(raw.data().end(), data, data + size);
-	} else {
-		raw.data().resize(new_data_size);
-		memcpy(raw.data().data() + io->offset, data, size);
+	if (update_json) {
+		if (request.json.size()) {
+			it->json()->assign(reinterpret_cast<char *>(request.json.data()), request.json.size());
+		} else {
+			it->json()->clear();
+		}
+
+		if (cmd->cmd == DNET_CMD_WRITE_NEW) {
+			it->set_json_timestamp(request.json_timestamp);
+		} else {
+			it->clear_json_timestamp();
+		}
+	}
+
+	if (update_data) {
+		if (append) {
+			raw->append(reinterpret_cast<char *>(request.data.data()), request.data.size());
+		} else {
+			raw->resize(new_data_size);
+			raw->replace(request.data_offset, std::string::npos,
+				     reinterpret_cast<char *>(request.data.data()), request.data.size());
+		}
 	}
 	TIMER_STOP("write.modify");
 	m_cache_stats.size_of_objects += it->size();
 
 	it->set_remove_from_cache(false);
 	insert_data_into_page(id, new_page_number, &*it);
 
-	// Mark data as dirty one, so it will be synced to the disk
+	if (update_data) {
```

Comment:
I think that write with `DNET_IO_FLAGS_UPDATE_JSON` should also update **synctime** and **lifetime**.